xwayland.c

```diff
@@ -141,6 +150,9 @@ handle_xwayland_surface_destroy(struct wl_listener *listener, void *data)
 
 	wl_list_remove(&xwayland_view->destroy.link);
 	wl_list_remove(&xwayland_view->request_fullscreen.link);
+	wl_list_remove(&xwayland_view->associate.link);
+	wl_list_remove(&xwayland_view->dissociate.link);
```

Comment:
Ah, good catch! I noticed the same missing calls in https://github.com/cage-kiosk/cage/pull/459. Could we keep this bugfix separate from support for a new protocol?